src/unwrap.jl

```diff
@@ -97,38 +97,39 @@ end
 Pixel(v, rng) = Pixel{typeof(v)}(0, v, rand(rng), 1)
 @inline Base.length(p::Pixel) = p.head.groupsize
 
-struct Edge{N}
+struct Edge{T}
     reliability::Float64
     periods::Int
-    pixel_1::CartesianIndex{N}
-    pixel_2::CartesianIndex{N}
+    pixel_1::Pixel{T}
+    pixel_2::Pixel{T}
 end
-function Edge{N}(pixel_image::AbstractArray, ind1::CartesianIndex{N}, ind2::CartesianIndex{N}, range) where N
-    @inbounds rel = pixel_image[ind1].reliability + pixel_image[ind2].reliability
-    @inbounds periods = find_period(pixel_image[ind1].val, pixel_image[ind2].val, range)
-    return Edge{N}(rel, periods, ind1, ind2)
+function Edge{T}(p1::Pixel{T}, p2::Pixel{T}, range) where {T}
+    rel = p1.reliability + p2.reliability
+    periods = find_period(p1.val, p2.val, range)
+    return Edge{T}(rel, periods, p1, p2)
 end
 @inline Base.isless(e1::Edge, e2::Edge) = isless(e1.reliability, e2.reliability)
 
 function unwrap_nd!(dest::AbstractArray{T, N},
                     src::AbstractArray{T, N};
                     range::Number=2*convert(T, pi),
                     circular_dims::NTuple{N, Bool}=ntuple(_->false, Val(N)),
-                    rng::AbstractRNG=GLOBAL_RNG) where {T, N}
+                    rng::AbstractRNG=default_rng()) where {T, N}
 
     range_T = convert(T, range)
 
     pixel_image = init_pixels(src, rng)
     calculate_reliability(pixel_image, circular_dims, range_T)
-    edges = Edge{N}[]
+    edges = Edge{T}[]
     num_edges = _predict_num_edges(size(src), circular_dims)
     sizehint!(edges, num_edges)
-    for idx_dim=1:N
+    for idx_dim = 1:N
         populate_edges!(edges, pixel_image, idx_dim, circular_dims[idx_dim], range_T)
     end
 
-    sort!(edges, alg=MergeSort)
-    gather_pixels!(pixel_image, edges)
+    perm = sortperm(map(x -> x.reliability, edges); alg=MergeSort)
+    edges = edges[perm]
```

Comment:
Why is this better? And why the `map`? Shouldn't the custom `isless` take care of that? And if not, wouldn't a `by=` be better by avoiding a temporary array?

Reply:
> I'm not sure why, some cache locality issues when sorting? Because Edges are 4 times the size of floats.

Ah yes, that makes sense. Less data movement by doing `sortperm` and then indexing.



> There are also allocations elsewhere that could be reduced, although I feel that would obscure the code.

Right, but
```julia
perm = sortperm(edges; by=x -> x.reliability, alg=MergeSort)
```
isn't exactly more obscure? Or even `perm = sortperm(edges; alg=MergeSort)`, but relying on the `isless` method might be considered a bit obscure. Anyway, my approval to this PR persists, so feel free to merge if you don't want change this; it's really not important.

Reply:
Ok, whatever... Merge as is, then?